src/components/related-assets/related-assets.tsx

```diff
@@ -1,58 +1,65 @@
+import { RelatedAssetGroup } from "./related-asset-group";
+import { RelatedAsset } from "./types";
+import { getAllRelatedAssets, organiseRelatedAssetsByType } from "./utils";
 
-import { RelatedAsset as IRelatedAsset } from "views/related-assets-view/utils";
+import { Asset, ParentAsset } from "@mtfh/common/lib/api/asset/v1";
+import { Center, Spinner } from "@mtfh/common/lib/components";
 
-export interface RelatedAssetsProps {
-  assetType: string;
-  relatedAssets: IRelatedAsset[];
+interface RelatedAssetsProps {
+  parentAssets: ParentAsset[];
+  childrenAssets: Asset[] | undefined;
+  loading: boolean;
 }
 
-export const RelatedAssets = ({ assetType, relatedAssets }: RelatedAssetsProps) => {
-  const getAssetTypeHeading = () => {
-    return assetType.charAt(assetType.length - 1).toLowerCase() === "s"
-      ? assetType
-      : `${assetType}s`;
-  };
+export const RelatedAssets = (props: RelatedAssetsProps) => {
+  const { parentAssets, childrenAssets, loading } = props;
 
-  const renderRelatedAssets = () => {
-    return relatedAssets.map((relatedAsset) => {
-      return <RelatedAsset relatedAsset={relatedAsset} key={relatedAsset.id} />;
-    });
-  };
+  const [relatedAssets, setRelatedAssets] = useState<RelatedAsset[]>([]);
+  const [relatedAssetsByType, setRelatedAssetsByType] = useState<any>(undefined);
 
-  return (
-    <section>
-      <p className="lbh-body-m">{getAssetTypeHeading()}</p>
-      <div className="mtfh-card-list" data-testid="related-asset-items-group">
-        {renderRelatedAssets()}
-      </div>
-    </section>
-  );
-};
+  useEffect(() => {
+    if (parentAssets?.length && childrenAssets) {
+      setRelatedAssets(getAllRelatedAssets(parentAssets, childrenAssets));
+    }
+  }, [parentAssets, childrenAssets]);
 
-export interface RelatedAssetProps {
-  relatedAsset: IRelatedAsset;
-}
+  useEffect(() => {
+    if (relatedAssets.length) {
+      const assetsByType = organiseRelatedAssetsByType(relatedAssets);
+
+      setRelatedAssetsByType(assetsByType);
+    }
+  }, [relatedAssets]);
+
+  if (loading) {
+    return (
+      <Center>
+        <Spinner />
+      </Center>
+    );
+  }
+
+  if (relatedAssets.length || !relatedAssetsByType) {
+    return (
+      <p className="lbh-body-m" data-testid="no-related-assets-message">
+        There are no related assets for this property.
+      </p>
+    );
+  }
 
-export const RelatedAsset = ({ relatedAsset }: RelatedAssetProps) => {
   return (
-    <LinkBox data-testid="related-asset">
-      <SearchCard>
-        <LinkOverlay>
-          <Link
-            className="mtfh-search-tenure__title"
-            as={RouterLink}
-            to={`/property/${relatedAsset.id}`}
-            variant="text-colour"
-          >
-            {relatedAsset.name}
-          </Link>
-        </LinkOverlay>
-      </SearchCard>
-    </LinkBox>
+    <>
+      {Object.keys(relatedAssetsByType).map((assetType) => (
+        <React.Fragment key={assetType}>
```

Comment:
Can you replace "React.Fragment" with a "div"? I thought I had removed this, earlier this morning after your comments, not sure why it's back